datajunction-server/datajunction_server/api/nodes.py

```diff
@@ -1169,7 +1172,7 @@ def create_a_source(
         if not query_service_client:
             raise DJException(
                 message="No table columns were provided and no query "
-                "service is configured for table columns inference!",
+                "service is configured for table columns inference",
```

Comment:
Should the `create_a_source` endpoint be modified to only work for source nodes where the columns are manually managed? Then it's very clear that in one case the columns are managed via a reflection/query service, but in the other case the columns are specified manually and there's no need for the server to attempt to resolve between the two cases.

Reply:
Updated this and also made columns required for the manual source node creation path through `/nodes/source/`. It seems pretty clear and intuitive now. Something we might want to eventually add is a check across all node creation to make sure nobody creates nodes in the `SOURCE_NODE_NAMESPACE`. I doubt this will be that big of a problem but it'll lead to cleaner reflection logic in the query service.